src/makeCalculator.js

```diff
@@ -3,8 +3,51 @@
 /**
  * @return {object}
  */
+
+/* eslint-disable no-console */
+
 function makeCalculator() {
-  // write code here
+  const calculator = {
+    result: 0,
+
+    add(number) {
+      this.result += number;
+
+      return this;
+    },
+
+    subtract(number) {
+      this.result -= number;
+
+      return this;
+    },
+
+    multiply(number) {
+      this.result *= number;
+
+      return this;
+    },
+
+    divide(number) {
+      this.result /= number;
+
+      return this;
+    },
+
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+
+    operate(callback, number) {
+      callback.call(this, number);
```

Comment:
The `operate` method correctly uses the `call` method to set the appropriate `this` context for the callback function.